py/genv/remote/ssh.py

```diff
@@ -102,24 +63,21 @@ async def run(
 
     :return: Returns the hosts that succeeded and their standard outputs
     """
-    processes = [
-        await start(
-            host,
-            command,
-            asyncio.subprocess.PIPE if stdins else asyncio.subprocess.DEVNULL,
-        )
-        for host in config.hosts
-    ]
+    ssh_runners_and_inputs = []
+    for host, stdin in zip(config.hosts, stdins or [None for _ in config.hosts]):
+        ssh_runner = SshRunner(host.hostname, host.timeout, {'PATH': SshRunner.calc_remote_path_env(host.root)})
+        ssh_runners_and_inputs.append((ssh_runner, stdin))
 
-    outputs = await asyncio.gather(
+    ssh_outputs = await asyncio.gather(
         *(
-            process.communicate(stdin.encode("utf-8") if stdin else None)
-            for process, stdin in zip(processes, stdins or [None for _ in config.hosts])
+            ssh_runner.run("genv", *command.args, stdin=stdin, sudo=command.sudo)
+            for ssh_runner, stdin in ssh_runners_and_inputs
         )
     )
 
-    stdouts = [stdout.decode("utf-8").strip() for stdout, _ in outputs]
-    stderrs = [stderr.decode("utf-8").strip() for _, stderr in outputs]
+    processes = [process for process, _, _, _ in ssh_outputs]
```

Comment:
is this used?
if not, should we actually return `Process` from `run()`?
what about `command`, should we return it?